Snowflake.Data.Tests/UnitTests/SFStatementTest.cs

```diff
@@ -118,9 +119,93 @@ public void TestTrimSqlLineCommentWithClosingNewline()
             SFSession sfSession = new SFSession("account=test;user=test;password=test", null, restRequester);
             sfSession.Open();
             SFStatement statement = new SFStatement(sfSession);
-            SFBaseResultSet resultSet = statement.Execute(0, "--comment\r\nselect 1\r\n--comment\r\n", null, false);
+            SFBaseResultSet resultSet = statement.Execute(0, "--comment\r\nselect 1\r\n--comment\r\n", null, false, false);
             Assert.AreEqual(true, resultSet.Next());
             Assert.AreEqual("1", resultSet.GetString(0));
         }
+
+        [Test]
+        [TestCase("UNKNOWN", QueryStatus.Unknown)]
+        [TestCase("RANDOM_STATUS", QueryStatus.Unknown)]
+        [TestCase("aBcZyX", QueryStatus.Unknown)]
+        [TestCase("running", QueryStatus.Running)]
+        [TestCase("RUNNING", QueryStatus.Running)]
+        [TestCase("resuming_warehouse", QueryStatus.ResumingWarehouse)]
+        [TestCase("RESUMING_WAREHOUSE", QueryStatus.ResumingWarehouse)]
+        [TestCase("queued", QueryStatus.Queued)]
+        [TestCase("QUEUED", QueryStatus.Queued)]
+        [TestCase("queued_reparing_warehouse", QueryStatus.QueuedReparingWarehouse)]
+        [TestCase("QUEUED_REPARING_WAREHOUSE", QueryStatus.QueuedReparingWarehouse)]
+        [TestCase("no_data", QueryStatus.NoData)]
+        [TestCase("NO_DATA", QueryStatus.NoData)]
+        [TestCase("aborting", QueryStatus.Aborting)]
+        [TestCase("ABORTING", QueryStatus.Aborting)]
+        [TestCase("success", QueryStatus.Success)]
+        [TestCase("SUCCESS", QueryStatus.Success)]
+        [TestCase("failed_with_error", QueryStatus.FailedWithError)]
+        [TestCase("FAILED_WITH_ERROR", QueryStatus.FailedWithError)]
+        [TestCase("aborted", QueryStatus.Aborted)]
+        [TestCase("ABORTED", QueryStatus.Aborted)]
+        [TestCase("failed_with_incident", QueryStatus.FailedWithIncident)]
+        [TestCase("FAILED_WITH_INCIDENT", QueryStatus.FailedWithIncident)]
+        [TestCase("disconnected", QueryStatus.Disconnected)]
+        [TestCase("DISCONNECTED", QueryStatus.Disconnected)]
+        [TestCase("restarted", QueryStatus.Restarted)]
+        [TestCase("RESTARTED", QueryStatus.Restarted)]
+        [TestCase("blocked", QueryStatus.Blocked)]
+        [TestCase("BLOCKED", QueryStatus.Blocked)]
+        public void TestGetQueryStatusByStringValue(string status, QueryStatus expectedStatus)
```

Comment:
let's split it into 2 tests: one for positive cases and one for negative ones (errors)

Reply:
Done